### PR TITLE
fix: issue import export asynchronous

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
make import-export issue excel async

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=263103&issueFilter__urlQuery=eyJ0aXRsZSI6IuOAkOaAp%2BiDveS8mOWMluOAkeWvvOWFpeWvvOWHuuS8mOWMliIsInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl19&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&iterationID=772&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | make import-export issue excel async              |
| 🇨🇳 中文    |   性能优化，导入导出事项异步           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Effet** [comment]
we should separate upload and import as two apis

**Comment by chengjoey** [comment]
> we should separate upload and import as two apis

changed，import issue by file uuid

File: apistructs/issue.go [hunk 1]
```diff
@@ -682,14 +682,16 @@ func (ipr *IssuePagingRequest) GetUserIDs() []string {
...
 
 // IssueImportExcelRequest 事件导入excel请求
 type IssueImportExcelRequest struct {
-	ProjectID uint64            `json:"projectID"`
-	OrgID     int64             `json:"orgID"`
-	Type      PropertyIssueType `json:"type"`
+	ProjectID   uint64            `json:"projectID"`
+	OrgID       int64             `json:"orgID"`
+	Type        PropertyIssueType `json:"type"`
+	ApiFileUUID string            `json:"apiFileUUID"`
```
Comment (Effet) [inline apistructs/issue.go:694]: you should simplify to `FileID`

File: apistructs/issue.go [hunk 2]
```diff
@@ -682,14 +682,16 @@ func (ipr *IssuePagingRequest) GetUserIDs() []string {
...
 }
 
 // IssueImportExcelRequest 事件导入excel请求
 type IssueImportExcelRequest struct {
-	ProjectID uint64            `json:"projectID"`
-	OrgID     int64             `json:"orgID"`
-	Type      PropertyIssueType `json:"type"`
+	ProjectID   uint64            `json:"projectID"`
+	OrgID       int64             `json:"orgID"`
+	Type        PropertyIssueType `json:"type"`
```
Comment (Effet) [inline apistructs/issue.go:693]: type should change to issueType

File: apistructs/issue.go [hunk 3]
```diff
@@ -763,7 +765,7 @@ type IssueListRequest struct {
 	// internal use, get from *http.Request
 	IdentityInfo
 	// 用来区分是通过ui还是bundle创建的
-	External bool `json:"-"`
+	External bool `json:"external"`
```
Comment (Effet) [inline apistructs/issue.go:768]: will it break down other logic?

File: modules/dop/services/issue/issue.go [hunk 4]
```diff
@@ -25,6 +25,8 @@ import (
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 
+	"github.com/erda-project/erda/modules/dop/services/issueproperty"
+
```
Comment (Effet) [inline modules/dop/services/issue/issue.go:29]: empty line

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/services/issue/issue.go (hunk 4 above)
Comment (chengjoey) [reply modules/dop/services/issue/issue.go]: adjusted

File: apistructs/issue.go (hunk 2 above)
Comment (chengjoey) [reply apistructs/issue.go]: this filed will be quote by others and front engineers use `propertyIssueType`, so keep PropertyIssueType
image

File: apistructs/issue.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/issue.go]: adjusted

File: modules/dop/services/issue/export.go [hunk 5]
```diff
@@ -34,6 +38,92 @@ type issueStage struct {
...
+	req.PageSize = 99999
+	fileReq := apistructs.TestFileRecordRequest{
+		ProjectID:    req.ProjectID,
+		OrgID:        uint64(req.OrgID),
+		Type:         apistructs.FileIssueActionTypeExport,
+		State:        apistructs.FileRecordStatePending,
+		IdentityInfo: req.IdentityInfo,
+		Extra: apistructs.TestFileExtra{
+			IssueFileExtraInfo: &apistructs.IssueFileExtraInfo{
+				ExportRequest: req,
```
Comment (sfwn) [inline modules/dop/services/issue/export.go:52]: pageSize just for record or really used?

File: modules/dop/services/issue/issue.go [hunk 6]
```diff
@@ -45,6 +46,10 @@ type Issue struct {
 	uc           *ucauth.UCClient
 	tran         i18n.Translator
 	issueRelated *issuerelated.IssueRelated
+	Ip           *issueproperty.IssueProperty
```
Comment (sfwn) [inline modules/dop/services/issue/issue.go:49]: bad abbreviation

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/services/issue/export.go (hunk 5 above)
Comment (chengjoey) [reply modules/dop/services/issue/export.go:52]: really used

File: modules/dop/services/issue/issue.go (hunk 6 above)
Comment (chengjoey) [reply modules/dop/services/issue/issue.go]: adjusted

File: modules/dop/services/issue/import.go [hunk 7]
```diff
@@ -17,29 +17,115 @@ package issue
...
 	"github.com/erda-project/erda/modules/dop/services/issueproperty"
 	"github.com/erda-project/erda/pkg/database/dbengine"
 )
 
-func (svc *Issue) ImportExcel(req apistructs.IssueImportExcelRequest, r *http.Request, properties []apistructs.IssuePropertyIndex, ip *issueproperty.IssueProperty, member []apistructs.Member) (*apistructs.IssueImportExcelResponse, error) {
-	// 获取测试用例数据
-	f, _, err := r.FormFile("file")
+const (
+	fileUploadFrom     = "autotest-space"
+	fileValidityPeriod = 7 * time.Hour * 24
```
Comment (Effet) [inline modules/dop/services/issue/import.go:32]: make it configurable

**Review by Effet — DISMISSED** [review]

File: modules/dop/services/issue/import.go (hunk 7 above)
Comment (chengjoey) [reply modules/dop/services/issue/import.go]: adjusted, remove this configuration, becasue it's decided by front now
i made export issue file validity-period configurable
image

**Review by sfwn — APPROVED** [review]

**Pull request merged**